PRO58/Src/main.cpp

```diff
@@ -79,267 +83,276 @@ void SystemClock_Config(void);
 
 /* USER CODE BEGIN PFP */
 /* Private function prototypes -----------------------------------------------*/
-static void globalMenuButtonHandler(
-  Button button,
-  Buttons::PressType pressType
-);
+static void globalMenuButtonHandler(Button button,
+		Buttons::PressType pressType);
 /* USER CODE END PFP */
 
 /* USER CODE BEGIN 0 */
 
 /* USER CODE END 0 */
 
-int main(void)
-{
+int main(void) {
+
+	/* USER CODE BEGIN 1 */
+
+	/* USER CODE END 1 */
+
+	/* MCU Configuration----------------------------------------------------------*/
+
+	/* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+	HAL_Init();
+
+	/* USER CODE BEGIN Init */
+
+	/* USER CODE END Init */
+
+	/* Configure the system clock */
+	SystemClock_Config();
+
+	/* USER CODE BEGIN SysInit */
+	HAL_Delay(200);
+	/* USER CODE END SysInit */
+
+	/* Initialize all configured peripherals */
+	MX_GPIO_Init();
+	MX_DMA_Init();
+	MX_ADC1_Init();
+	MX_I2C2_Init();
+	MX_USART1_UART_Init();
+#ifndef HB5808
+	MX_I2C1_Init();
+#endif
+#ifdef USE_BUZZER
+	MX_TIM4_Init();
+#endif
+	MX_TIM3_Init();
+
+	/* USER CODE BEGIN 2 */
+	if (DWT_Delay_Init()) {
+		_Error_Handler(__FILE__, __LINE__); /* Call Error Handler */
+	}
+
+	HAL_GPIO_WritePin(RECEIVER_SW_GPIO_Port, RECEIVER_SW_Pin, GPIO_PIN_RESET);
+	HAL_GPIO_WritePin(SPI_SLAVE_SELECT_A_GPIO_Port, SPI_SLAVE_SELECT_A_Pin,
+			GPIO_PIN_SET);
+	HAL_GPIO_WritePin(SPI_SLAVE_SELECT_B_GPIO_Port, SPI_SLAVE_SELECT_B_Pin,
+			GPIO_PIN_SET);
+	HAL_GPIO_WritePin(SPI_CLOCK_GPIO_Port, SPI_CLOCK_Pin, GPIO_PIN_RESET);
+	HAL_GPIO_WritePin(SPI_DATA_GPIO_Port, SPI_DATA_Pin, GPIO_PIN_RESET);
+
+	HAL_Delay(100); //Delay 1000ms to allow RX5808 startup.
+
+#ifndef HB5808
+	I2C_Reset(hi2c1, MX_I2C1_Init);
+	Ui::setup(&hi2c1);
+#else
+	I2C_Reset(hi2c2, MX_I2C2_Init);
+	Ui::setup(&hi2c2);
+#endif
+
+	Ui::display.drawBitmap(0, 0, logo,
+	SCREEN_WIDTH,
+	SCREEN_HEIGHT,
+	WHITE);
+	Ui::display.display();
+
+#ifdef USE_BUZZER
+	Beeper::init();
```

Comment:
Maybe move the init/welcome beep sequence into beeper, would clean up the main a bit

Reply:
Good idea